curriculum/challenges/english/10-coding-interview-prep/rosetta-code/self-referential-sequence.english.md

```diff
@@ -0,0 +1,152 @@
+---
+id: 5eb3e4a21f462f409d656c73
+title: Self-referential sequence
+challengeType: 5
+isHidden: false
+forumTopicId: 385317
+---
+
+## Description
+
+<section id='description'>
+There are several ways to generate a self-referential sequence. One very common one (the <a href="https://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:
+
+<pre>0, 10, 1110, 3110, 132110, 1113122110, 311311222110 ...</pre>
+
+The terms generated grow in length geometrically and never converge.
+
+Another way to generate a self-referential sequence is to summarize the previous term.
+
+Count how many of each alike digit there is, then concatenate the sum and digit for each of the sorted enumerated digits. Note that the first five terms are the same as for the previous sequence.
+
+<pre>0, 10, 1110, 3110, 132110, 13123110, 23124110 ...</pre>
+
+Sort the digits largest to smallest. Do not include counts of digits that do not appear in the previous term.
+
+Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, converge means that an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after the first element. (The next element is 22, which has been seen before.)
+</section>
+
+## Instructions
+
+<section id='instructions'>
+Write a function that takes the seed value as parameter and returns the self referential sequence as an array.
```

Comment:
@moT01 I think the second method is the one in which the series converges and hence it is the focus of the task. For the first method, it is stated: "The terms generated grow in length geometrically and never converge". As it is an infinite sequence, it cannot have a finite answer. I also think that we can make the instructions more clear.
```
Write a function that takes the seed value as parameter, generate a  self referential sequence until it converges and return it as an array.
```
This would imply to use the second method as the first one do not converge.

Reply:
@NitronR, thanks for improving these instructions. Your proposal sounds much clearer. I just want to suggest a couple of small changes:

`Write a function that takes the seed value as parameter, generates a self referential sequence until it converges, and returns it as an array.`